src/fmi4c_utils.c

```diff
@@ -21,27 +21,35 @@
 #endif
 
 
-void rememberPointer(fmiHandle *fmu, int* ptr) {
-
+void rememberPointer(fmiHandle *fmu, void* ptr)
+{
     fmu->numAllocatedPointers++;
     fmu->allocatedPointers = realloc(fmu->allocatedPointers, fmu->numAllocatedPointers * sizeof(void*));
     fmu->allocatedPointers[fmu->numAllocatedPointers-1] = ptr;
 }
 
-void* mallocAndRememberPointer(fmiHandle *fmu, size_t size) {
+void* mallocAndRememberPointer(fmiHandle *fmu, size_t size)
+{
     void* ptr = malloc(size);
     rememberPointer(fmu, ptr);
     return ptr;
 }
 
 void *reallocAndRememberPointer(fmiHandle *fmu, void *org, size_t size)
 {
+    int i=0;
+    while (org != fmu->allocatedPointers[i] && i < fmu->numAllocatedPointers)
```

Comment:
The conditions should be swapped. Even thought the second condition should not even be necessary. The code just looked dangerous with this potential infinite loop, which is why I added it.

Reply:
The second argument is good in case someone calls this function for a pointer that was malloced but not remembered (which should not be done, but still...). Do you want to swap the conditions or can I merge?

Reply:
It must be swapped; otherwise, it will access memory out of bounds.